hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieDeleteBlock.java

```diff
@@ -86,7 +86,7 @@ public HoodieDeleteBlock(Option<byte[]> content, FSDataInputStream inputStream,
         int dataLength = dis.readInt();
         byte[] data = new byte[dataLength];
         dis.readFully(data);
-        this.keysToDelete = SerializationUtils.<HoodieKey[]>deserialize(data);
+        this.keysToDelete = SerializationUtils.<DeleteKey[]>deserialize(data);
```

Comment:
This is going to be non-BWC. 

To avoid surprises i think we should do either of
 - Support both log-formats (adding new format version for deletion blocks)
 - Make sure we can enforce proper upgrading sequence (first compact, then upgrade)
 - Make it BWC

Reply:
1 & 3 is okey to me, @xushiyan would follow this in another PR then ~